org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/AcquisitionDevice.java

```diff
@@ -208,14 +200,74 @@ public void configure(ScanModel model) throws ScanningException {
 		setConfigureTime(after-before);
 	}
 
+	private void setScannables(ScanModel model) throws ScanningException {
+		List<IScannable<?>> scannables = model.getScannables();
+		if (scannables == null) {
+			scannables = connectorService.getScannables(model.getPositionIterable());
+		}
+		
+		// calculate the names of the axes controlled by malcolm device(s)
+		final Set<String> malcolmControlledAxes = model.getDetectors().stream().
+				filter(IMalcolmDevice.class::isInstance).
+				map(IMalcolmDevice.class::cast).
+				flatMap(m -> getAxesToMove(m).stream()).
+				collect(Collectors.toSet());
+		
+		scannables = scannables.stream().
+				filter(s -> !malcolmControlledAxes.contains(s.getName())).
+				collect(Collectors.toList());
+		
+		model.setScannables(scannables);
+	}
+	
+	private Set<String> getAxesToMove(IMalcolmDevice<?> malcolmDevice) {
+		try {
+			return ((IMalcolmDevice<?>) malcolmDevice).getAxesToMove();
+		} catch (ScanningException e) {
+			throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)